Deliverers' ID attribute recognised by Specberus, and spat out as part of metadata

tripu opened the pull request:
We haven't reached consensus in #130. But @deniak and I favoured reading a new attribute `data-deliverer-id` on documents, and it's the quickest solution for the issue. So here is that solution.

For more info about how those IDs are being returned now, see [the updated documentation](https://github.com/w3c/specberus/blob/tripu/issue-130-deliverers-metadata/README.md#emitting-metadata-about-the-document).

Also, notice that the new attribute **is not required** by Specberus, ie not finding it on the document will *not* throw an error.

astorije commented:
@deniak, @tripu and I discussed this on the phone today. We will use @dontcallmedom's suggestion to extract the WG ID from the PP link, to get to the point faster, but some of us express concerns about this solution. @plh had ideas and we will most likely refactor after v1.

@tripu, I will review when you are done, no rush :)

tripu commented:
@astorije: change done; please review.

astorije commented on README.md:
```diff
@@ -84,11 +84,14 @@ These properties are now returned when found:
 * `previousVersion`: URL of the previous version of the document (the last one, if multiple are shown).
 * `latestVersion`: URL of the latest version of the document.
-* `editorIDs`: ID(s) of the editor(s) responsible for the document, necessarily matching the pattern `/\d+/`.
+* `editorIDs`: ID(s) of the editor(s) responsible for the document (an `Array` of `String`s, necessarily matching the pattern `/\d+/`).
```
Why do you need to specify that this is an `Array` of `String`s? Isn't it possible to send an array of integers instead?

astorije commented on README.md:
```diff
@@ -84,11 +84,14 @@ These properties are now returned when found:
 * `latestVersion`: URL of the latest version of the document.
-* `editorIDs`: ID(s) of the editor(s) responsible for the document, necessarily matching the pattern `/\d+/`.
+* `editorIDs`: ID(s) of the editor(s) responsible for the document (an `Array` of `String`s, necessarily matching the pattern `/\d+/`).
+* `delivererID` ID of the group, (a `String`, necessarily matching the pattern `/\d+/`).
```
Why isn't this part of the `deliverers`? Something like:
- `deliverers`
  - `id`
  - `homepage`
  - `name`

Also, there can be multiple deliverers, so there will be multiple IDs detected.

astorije commented on lib/rules/heuristic/group.js:
```diff
@@ -4,16 +4,28 @@
+    ,   candidates = []
+    ,   candidate
+    ,   item
+    ,   i;
```
This is such an ugly, error-prone syntax! (I am not blaming you but the very first guy who used it :) )

FYI, I plan to switch them back all to the traditional `var [...];` one. This horror has to stop and the world needs a leader :D

astorije commented on lib/rules/sotd/pp.js:
```diff
@@ -64,17 +70,25 @@ exports.check = function (sr, done) {
+            text === "public list of any patent disclosures") {
+            ids = DELIVERER_ID_REGEX.exec(href);
+            console.log (href + ' - ' + DELIVERER_ID_REGEX);
+            console.dir(ids);
```
Aren't these 2 lines remains from your development tests? If not, how useful is it in general? This is not sarcasm this time, I am genuinely wondering :)

astorije commented:
@tripu, review done. Mostly, a bit puzzled about the `delivererID` scalar param versus the `deliverers` collection param.

tripu commented:
@astorije, [I've committed a few changes](https://github.com/w3c/specberus/commit/09c4cafd06f001d89a729f61e9efcf3e2c862c8e), in response to your comments.
Details:

* `delivererIDs` and `editorIDs` are now `Array`s of `Number`s (not `String`s like before).
* Now, multiple `delivererIDs` are returned (it's a vector instead of an scalar). Those IDs can't be part of the `Object`s inside `deliverers` because those two pieces of information are extracted in different ways, from different parts of the document. There's no (consistent) way to associated IDs to names and homepages: there could be a different number of IDs and homepages, and even if the number is the same, how do we know which is which&hellip;?
* Removed `console.log()` and `console.dir()` statements (oops! thanks!).

Additional fixes:

* `README.md`: updated with latest changes to metadata fields, fixed hierarchy of document profiles, fixed typo, improved styling.
* Profile `TR`: fixed a bug (rule `sotd/pp` was being checked twice).

astorije commented:
Looks good, I just suggest a fix in #135.
